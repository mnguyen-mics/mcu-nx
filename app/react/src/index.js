import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
