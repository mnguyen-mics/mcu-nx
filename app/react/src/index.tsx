/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
