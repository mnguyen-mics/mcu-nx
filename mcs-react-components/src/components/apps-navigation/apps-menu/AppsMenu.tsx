import * as React from 'react';
import { Menu } from 'antd';

export type AppsMenuSections = {
  adminLinks: AppsMenuItem[];
  userLinks: AppsMenuItem[];
  resourceLinks?: AppsMenuItem[];
};
export interface AppsMenuItem {
  icon?: React.ReactElement;
  name: string;
  url: string;
}

export interface AppsMenuProps {
  logo: React.ReactElement;
  sections: AppsMenuSections;
  className?: string;
}

class AppsMenu extends React.Component<AppsMenuProps> {
  constructor(props: AppsMenuProps) {
    super(props);
    this.state = {};
  }

  renderItem(item: AppsMenuItem, displayBorder: boolean): React.ReactElement {
    return (
      <Menu.Item
        icon={item.icon}
        className={
          displayBorder ? 'mcs_appMenu_item mcs_appMenu_item--withDivider' : 'mcs_appMenu_item'
        }
      >
        <a href={item.url}>
          <span>{item.name}</span>
        </a>
      </Menu.Item>
    );
  }

  renderSection(links: AppsMenuItem[], displayBorder?: boolean): React.ReactElement[] {
    return links.map((link, index) =>
      this.renderItem(link, !!displayBorder && links.length === index + 1),
    );
  }

  render() {
    const { sections, logo, className } = this.props;

    return (
      <Menu mode='inline' className={`mcs_appMenu ${className ? className : ''}`}>
        {logo}
        {sections.adminLinks && this.renderSection(sections.adminLinks, true)}
        {sections.userLinks && this.renderSection(sections.userLinks, true)}
        {sections.resourceLinks && this.renderSection(sections.userLinks)}
      </Menu>
    );
  }
}

export default AppsMenu;
