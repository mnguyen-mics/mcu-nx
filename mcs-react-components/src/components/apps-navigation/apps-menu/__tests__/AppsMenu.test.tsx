import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import AppsMenu, { AppsMenuProps } from '../AppsMenu';

it('render the AppsMenu', () => {
  const props: AppsMenuProps = {
    sections: [
      { items: [{ name: 'Platform Admin', url: 'url1' }] },
      { items: [{ name: 'Navigator', url: 'url2' }] },
    ],
    logo: <span />,
  };
  const component = TestRenderer.create(<AppsMenu {...props} />);

  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
