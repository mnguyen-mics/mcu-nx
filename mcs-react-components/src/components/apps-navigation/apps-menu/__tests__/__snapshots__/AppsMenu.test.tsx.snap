// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the AppsMenu 1`] = `
<ul
  className="ant-menu ant-menu-light mcs_appMenu fake-class-name ant-menu-root ant-menu-inline"
  onMouseEnter={[Function]}
  onTransitionEnd={[Function]}
  role="menu"
  style={Object {}}
>
  <div>
    toto
  </div>
  <li
    className="ant-menu-item ant-menu-item-only-child mcs_appMenu_item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <a
      href="url1"
    >
      <span>
        Platform Admin
      </span>
    </a>
  </li>
  <li
    className="ant-menu-item ant-menu-item-only-child mcs_appMenu_item mcs_appMenu_item--withDivider"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <a
      href="url2"
    >
      <span>
        computing console
      </span>
    </a>
  </li>
  <li
    className="ant-menu-item ant-menu-item-only-child mcs_appMenu_item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <a
      href="url2"
    >
      <span>
        Navigator2
      </span>
    </a>
  </li>
  <li
    className="ant-menu-item ant-menu-item-only-child mcs_appMenu_item mcs_appMenu_item--withDivider"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <a
      href="url2"
    >
      <span>
        Navigator3
      </span>
    </a>
  </li>
  <li
    className="ant-menu-item ant-menu-item-only-child mcs_appMenu_item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <a
      href="url2"
    >
      <span>
        Navigator2
      </span>
    </a>
  </li>
  <li
    className="ant-menu-item ant-menu-item-only-child mcs_appMenu_item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <a
      href="url2"
    >
      <span>
        Navigator3
      </span>
    </a>
  </li>
</ul>
`;
